python/GafferImageUI/ImageViewUI.py

```diff
@@ -45,6 +45,59 @@
 import GafferImage
 import GafferImageUI
 
+class DisplayTransformPlugValueWidget( GafferUI.PresetsPlugValueWidget ) :
+
+	def __init__( self, plug, **kw ) :
+
+		self.__gpuPlug = plug.parent()["displayTransformGPU"]
+
+		self.__plugDirtiedConnection = self.__gpuPlug.node().plugDirtiedSignal().connect( lambda p : Gaffer.WeakMethod( self._updateFromPlug )() )
+		GafferUI.PresetsPlugValueWidget.__init__( self, plug, **kw )
+
+	def _presetsMenuDefinition( self ) :
+		result = GafferUI.PresetsPlugValueWidget._presetsMenuDefinition( self )
+		result.append( "/OptionsDivider", { "divider" : True, "label" : "Options" } )
+		
+		## \todo : This should use a different checkbox icon to distinguish a toggle from selecting presets
+		## ( graphics/toggle(On/Off).png feels kinda nice ).  This corresponds to the exclusive/non-exclusive
+		## distinction made by Qt, but we don't want to muck about with creating action groups.  Tom's proposal
+		## is to instead just use icons on the menu items to show the checkboxes, rather than telling Qt that the
+		## menu item is a checkbox.  Going to leave that up to him
+		result.append( "GPU",
+			{
+				"command" : Gaffer.WeakMethod( self.__toggleGpu ),
+				"checkBox" : self.__gpuPlug.getValue(),
+				"shortCut" : "Alt+G",
+			}
+		)
+		return result
+
+	def __toggleGpu( self, v ) :
+		self.__gpuPlug.setValue( not self.__gpuPlug.getValue() )
+
+	## \todo : Duplicating this code is very ugly.  Need a better way to override the menu
+	##         button text
+	def _updateFromPlug( self ) :
+
+		self._PresetsPlugValueWidget__menuButton.setEnabled( self._editable() )
+
+		text = ""
+		if self.getPlug() is not None :
+			with self.getContext() :
+				presetName = Gaffer.NodeAlgo.currentPreset( self.getPlug() )
+
+			text = presetName or "Invalid"
+
+		if self.__gpuPlug.getValue():
+			text += " (GPU)"
+		else:
+			text += " (slow)"
```

Comment:
CPU?

Reply:
I don't much like how few pixel of difference there are between "GPU" and "CPU", especially since I was thinking it might be nice to make this text smaller, but I've gone with CPU for now, see what you think of it it actually runs on your system.

Reply:
Generally speaking, we always try to be consistent with pairings.  So grammatically speaking, it's the same type of word. In that, if one is describing 'what it runs on', then so should the other state. If you were to  go with `slow` - is an adjective describing the result, then GPU should change to something like `fast`. Though I think then it might be better to use `accurate` over `slow`, as it better helps the user understand why they'd choose one or the other.

It seems people are already using standard parenthesis in production as part of their names eg, from CS: `Default (show display without primary)`. We don't want to overload this title with info, and we want GPU to be the default right? As such, how about:

 - Use square brackets `[CPU]`
 - Only append this if on the CPU path, and leave it as-is for GPU. 

Then the common presentation is less cluttered, and its more apparent when you're on something custom? So it'd be:

`Default` or `Default [CPU]`

Reply:
People seem to want to know which is which. 

How about this. Keep the dropdown exactly as it was, add a small LUT cube icon next to it to the left. That presents a menu with the processing options in.  This fits the pattern we have in the SceneView (icons = menu) and ImageView ( icons = clickable ), it gets rid of the original concern that were abusing a combo box, and allows us mutually exclusive items (eg `GPU` and `CPU`), and to adapt the icon to reflect CPU/GPU state.

It also means that peoples customisations of the current widget (to display the actual profile in `Default`) will just keep on working too.